filters/include/pcl/filters/frustum_culling.h

```diff
@@ -144,14 +146,33 @@ namespace pcl
           throw PCLException ("Horizontal field of view should be between 0 and 180(excluded).",
             "frustum_culling.h", "setHorizontalFOV");
         }
-        hfov_ = hfov;
+        fov_left_bound_ = -hfov / 2;
+        fov_right_bound_ = hfov / 2;
+      }
+
+      /** \brief Set the horizontal field of view for the camera in degrees
+        * \param[in] fov_left_bound the left bound of horizontal field of view
+        * \param[in] fov_right_bound the right bound of horizontal field of view
+        */
+      void
+      setHorizontalFOV (float fov_left_bound, float fov_right_bound)
+      {
+        if (fov_left_bound <= -90 || fov_right_bound >= 90 || fov_left_bound >= fov_right_bound)
+        {
+          throw PCLException ("Horizontal field of view bounds should be between -90 and 90(excluded). "
+              "And left bound should be smaller than right bound.",
+            "frustum_culling.h", "setHorizontalFOV");
+        }
+        fov_left_bound_ = fov_left_bound;
+        fov_right_bound_ = fov_right_bound;
       }
 
       /** \brief Get the horizontal field of view for the camera in degrees */
-      float 
-      getHorizontalFOV () const
+      void
```

Comment:
I would prefer to keep the old function that returns a float, so that everyone who is currently using `setHorizontalFOV (float hfov)` still has the corresponding getter-function.
There, we could either just return `fov_right_bound_ - fov_left_bound_`, or first check whether the abs values of the bounds differ and print a warning if they do.

Reply:
Well, both overloads could be kept. The old one having `float func(void)` signature and the new one 'void func(float&, float&)'.